### PR TITLE
passdb/userdb: Default driver to name

### PR DESCRIPTION
**Pull request by sirainen**
DOV-6529

### CODE REVIEW

File: source/configuration_manual/authentication/password_databases_passdb.rst [hunk 1]
```diff
@@ -123,24 +123,28 @@ Passdb setting
...
-   setting.
+   Name of the passdb. The :dovecot_core:ref:`passdb` filter name refers to
+   this setting. If the :dovecot_core:ref:`passdb_driver` setting is empty,
+   the passdb_name is used as the driver. This allows doing e.g.:
+
+   .. code-block::
+
+   passdb passwd-file {
+     passwd_file_path = /etc/dovecot/passwd
+   }
```
Comment (cmouse) [inline source/configuration_manual/authentication/password_databases_passdb.rst:140]: indentation error

File: source/configuration_manual/authentication/user_databases_userdb.rst [hunk 2]
```diff
@@ -69,24 +69,28 @@ Userdb settings
...
-   setting.
+   Name of the userdb. The :dovecot_core:ref:`userdb` filter name refers to
+   this setting. If the :dovecot_core:ref:`userdb_driver` setting is empty,
+   the userdb_name is used as the driver. This allows doing e.g.:
+
+   .. code-block::
+
+   userdb passwd-file {
+     passwd_file_path = /etc/dovecot/passwd
+   }
```
Comment (cmouse) [inline source/configuration_manual/authentication/user_databases_userdb.rst:86]: indentation error

File: source/configuration_manual/authentication/nss.rst [hunk 3]
```diff
@@ -1,39 +0,0 @@
-.. _authentication-nss:
-
-=====
- NSS
```
Comment (cmouse) [inline source/configuration_manual/authentication/nss.rst:4]: you need to remove all the links pointing here too.

File: source/configuration_manual/authentication/password_databases_passdb.rst [hunk 4]
```diff
@@ -123,24 +123,28 @@ Passdb setting
...
-   to users.  The :dovecot_core:ref:`passdb` filter name refers to this
-   setting.
+   Name of the passdb. The :dovecot_core:ref:`passdb` filter name refers to
+   this setting. If the :dovecot_core:ref:`passdb_driver` setting is empty,
+   the passdb_name is used as the driver. This allows doing e.g.:
+
+   .. code-block::
+
+      passdb passwd-file {
+	passwd_file_path = /etc/dovecot/passwd
```
[the remark this answers is not in the document]
Comment (sirainen) [reply source/configuration_manual/authentication/password_databases_passdb.rst:139]: There are CI tests for it, so yes.

[the remark this answers is not in the document]
Comment (sirainen) [reply source/configuration_manual/authentication/password_databases_passdb.rst:139]: It's also used in a few places in documentation, although looks like not in passwd-file page.

[the remark this answers is not in the document]
Comment (sirainen) [reply source/configuration_manual/authentication/password_databases_passdb.rst:139]: Oh, no, it is also in passwd-file page documentation.

**Pull request merged**